firedrake/slate/preconditioners.py

```diff
@@ -45,6 +45,7 @@ def initialize(self, pc):
         prefix = pc.getOptionsPrefix() + "hybridization_"
         _, P = pc.getOperators()
         self.cxt = P.getPythonContext()
+        self.app_cxt = self.cxt.appctx
```

Comment:
I know you like `cxt` over `ctx`, but this feels somewhat wilful :).

In fact, why does this need to be a member variable at all, you use it once, in this function, so just write `self.cxt.appctx` at that point, no?